Create actor unlinker CLI task, Refs# 11753

sbreker opened the pull request:
CLI task that will update cases where a creator could be linked to
information object using inheritance.

If a descr slug is specified, all creators on this node + descendants
will be examined.

If an actor slug is specified, all descrs with this creator will be
examined.
- only this actor can be modified.
- in cases where more than one creator would have been unlinked, nothing
will be done.

If actor or descr is NOT specified, all nodes will be examined to see
if creators could be inherited instead.

- corrected short description inpromoteUserToAdminTask.class.php

sbreker committed:
Create actor unlinker CLI task, Refs# 11753

CLI task that will update cases where a creator could be linked to
information object using inheritance.

If a descr slug is specified, all creators on this node + descendants
will be examined.

If an actor slug is specified, all descrs with this creator will be
examined.
- only this actor can be modified.
- in cases where more than one creator would have been unlinked, nothing
will be done.

If actor or descr is NOT specified, all nodes will be examined to see
if creators could be inherited instead.

- corrected short description inpromoteUserToAdminTask.class.php

mcantelon commented on lib/task/tools/unlinkCreatorTask.class.php:
```diff
@@ -0,0 +1,214 @@
+        unset($event->actor);
+        $event->save();
+        // Delete the event record too if there aren't any dates/times on it.
+        if (null == $event->getPlace()->name && null == $event->date && null == $event->name && null == $event->description && null == $event->startDate && null == $event->endDate && null == $event->startTime && null == $event->endTime)
```
Making this multi-line might be good for code readability.

mcantelon commented on lib/task/tools/unlinkCreatorTask.class.php:
```diff
@@ -0,0 +1,214 @@
+    $this->log('Done!');
+  }
+
+  private function unlinkCreators($options = array())
```
This function's pretty long so could be could to break some logic out into other class methods.

mcantelon commented on lib/task/tools/unlinkCreatorTask.class.php:
```diff
@@ -0,0 +1,214 @@
+    $this->name = 'unlink-creators';
+    $this->briefDescription = 'Unlink creators from descriptions so creator inheritance can be used.';
+    $this->detailedDescription = <<<EOF
+Unlink creators from descriptions so creator inheritance can be used.
```
Could be good to expand on the functionality in more detail here.

mcantelon commented:
A couple of minor nitpicks, but otherwise looks good!

sbreker committed:
Create actor unlinker CLI task, Refs# 11753

Additional changes:
- add more detail to CLI task description
- split out criteria builder into separate func
- add line feeds to long 'if' statement in removeCreator()

mcantelon commented:
👍

MikeFE commented:
In the future please don't forget to use arBaseTask, not sfBaseTask.
